test/candidate_diff/candidate_diff_test.go

```diff
@@ -0,0 +1,70 @@
+package candidate_diff_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+	"github.com/openshift/osde2e-common/pkg/clients/ocm"
+	rosaprovider "github.com/openshift/osde2e-common/pkg/openshift/rosa"
+	"golang.org/x/sync/errgroup"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var _ = Describe("Gap Analysis - Candidate Cluster Diff", Ordered, func() {
+	// provision two clusters: one Y-1, one Y
+	// run gap analysis script
+	// fail if there is a difference
+
+	var (
```

Comment:
@jbpratt This [page](https://docs.ci.openshift.org/docs/architecture/ci-operator/#testing-with-an-existing-openshift-release) is what I was mentioning earlier about how a job triggered by ci-operator for a given nightly OCP build. Injects the build ID into environment as an environment variable (e.g. _RELEASE_IMAGE_LATEST_). Which from there osde2e current installer version selector picks up that and checks for it existing in OCM.

We should take this into account when this test suite runs from CI to pick up a nightly build to deploy as the second cluster. For the first cluster version, we could auto determine it based on the target version (Y-1) from the candidate channel/verify the version exists in OCM.